index.js

```diff
@@ -1,13 +1,41 @@
 const mongoose = require('mongoose');
-const Recipe = require('./models/Recipe.model'); // Import of the model Recipe from './models/Recipe.model.js'
-const data = require('./data.js'); // Import of the data from './data.js'
+const Recipe = require('./models/Recipe.model');
+const data = require('./data.js');
+const recipe = {
+  title: "Poulet caramel aux pistaches",
+  level: "Easy Peasy",
+  ingredients: ["poulet", "caramel", "pistaches"],
+  dishType: "Dish",
+  cuisine: "nawak"
+}
 
-// Connection to the database "recipeApp"
 mongoose
   .connect('mongodb://localhost/recipe-app-dev', {
     useCreateIndex: true,
     useNewUrlParser: true,
     useUnifiedTopology: true
   })
   .then(x => console.log(`Connected to Mongo! Database name: "${x.connections[0].name}"`))
-  .catch(err => console.error('Error connecting to mongo', err));
+  .catch(error => console.error('Error connecting to mongo', error));
+
+const createNewRecipe = Recipe.create(recipe)
+  .then(recipe => console.log(`The recipe ${recipe.title} has been saved!`))
+  .catch(error => console.error(`Oh no! An error happened while saving a new recipe: ${error}`));
+
+const importRecipes = Recipe.insertMany(data)
+  .then (data => console.log(`The recipe ${data.map(recipe => recipe.title)} has been saved!`))
+  .catch(error => console.error(`Ouch! An error happened while saving new recipes: ${error}`)); 
+
+const editRecipe = Recipe.updateOne({title: 'Rigatoni alla Genovese'}, {duration: 100})
+  .then (updatedRecipe => console.log(`${updatedRecipe.title} has been updated!`))
+  .catch(error => console.error(`Oops! An error happened while updating the recipe: ${error}`))
+
+const deleteRecipe = Recipe.deleteOne({title: 'Carrot Cake'})
+  .then (deletedRecipe => console.log(`${deletedRecipe.title} has been deleted!`))
+  .catch(error => console.log(`Ooh! An error happened while deleting the recipe: ${error}`))
+
+Promise.all([createNewRecipe, importRecipes, editRecipe, deleteRecipe])
+  .then(values => {
+    console.log(`Life is magic! All promises were filled: ${values}`);
+    mongoose.connection.close();})
+  .catch(error => console.error(`Sadness, something went wrong resolving all promises: ${error}`))
```

Comment:
All promises are asynchronous, so when your try to delete and edit your recipes, they might not have been created yet, so at the end they are not being updated.

Promise.all runs your promises simultaneously, not in the respective order they are in the array.

You could run Promise.all with the createNewRecipe and the importRecipes since they do not depend on each other, but you'd have to put the editRecipe / deleteRecipe only once the other 2 functions have been run, and then close the DB.
Good catch on the Promise.all though